### PR TITLE
[Hybrid Script] Unify the symbol tables to one; support `tvm.container.Array`

### PR DESCRIPTION
**Pull request by were**
Unify symbol tables.
Support `tvm.container.Array`.

### CODE REVIEW

**Comment by were** [comment]
@kevinthesun Can you take a look?
@tqchen Can you let this in after CI is done?

File: tests/python/unittest/test_hybrid_script.py [hunk 1]
```diff
@@ -538,6 +543,19 @@ def foo(a):
     a = tvm.placeholder((10, ), name='a')
     run_and_check(foo, [a])
 
+def test_const_range():
+    @tvm.hybrid.script
+    def foo(a, b):
+        c = output_tensor(a.shape, a.dtype)
+        for i in const_range(2):
```
Comment (kevinthesun) [inline tests/python/unittest/test_hybrid_script.py]: Can we iterate b wrt the length of it?

Comment (kevinthesun) [reply tests/python/unittest/test_hybrid_script.py]: Yes

Comment (were) [reply tests/python/unittest/test_hybrid_script.py]: This is done.

**Pull request merged**